c/tskit/tables.h

```diff
@@ -2708,19 +2727,33 @@ they appear in the ``nodes`` argument. Specifically, this subsets and reorders
 each of the tables as follows:
 
 1. Nodes: if in the list of nodes, and in the order provided.
-2. Individuals and Populations: if referred to by a retained node,
-   and in the order first seen when traversing the list of retained nodes.
-3. Edges: if both parent and child are retained nodes.
-4. Mutations: if the mutation's node is a retained node.
-5. Sites: if any mutations remain at the site after removing mutations.
+2. Individuals, if `TSK_FILTER_INDIVIDUALS`: if referred to by a retained node,
+    and in the order first seen when traversing the list of retained nodes.
+3. Populations, if `TSK_FILTER_POPULATIONS`: the same as Individuals.
+4. Edges: if both parent and child are retained nodes.
+5. Mutations: if the mutation's node is a retained node.
+6. Sites, if `TSK_FILTER_SITES`: if any mutations remain at the site after
+    removing mutations.
 
 Retained edges, mutations, and sites appear in the same
-order as in the original tables.
+order as in the original tables. If any of `TSK_FILTER_INDIVIDUALS`,
+`TSK_FILTER_POPULATIONS`, or `TSK_FILTER_SITES` are *not* provided,
+then the respective tables will be *unchanged*.
```

Comment:
Oh, you want the other tables entirely unchanged? Ah, *that's* what I've been confused about. Right, you don't want to reorder the populations because you want stable population/individual IDs. 

Well, what if we add a flag ``TSK_STABLE_IDs`` or something to ``subset`` then which does the least amount of ID reshuffling as possible, which still doing the basic node-subset operation? This could be mutually exclusive to the FILTER_X operations, or we could drop them entirely if they're not useful.

Or maybe your original idea of a  ``reorder_nodes`` function is better? I'm finding it hard to understand the desired semantics of the ``subset`` function at this point, so maybe another function is a better idea?